server/utils/socketAction.js

```diff
@@ -0,0 +1,16 @@
+const { emit } = require('../utils');
+
+const waitForAction = (socket, event, requestAction, receiveAction) => (
+  new Promise((resolve) => {
+    socket.once(event, async (payload) => {
```

Comment:
I'm a bit uncertain on how Node handles these promises if e.g. 1000000 users open and quickly closes a WS connection (as a DoS attempt). NodeJS runtime will GC this if it's not referenced and/or in scope, so it most likely won't be an issue. It could be something to consider researching further, though.

Reply:
I think it will be garbage collected like everything else in connection when socket is disconnected.